docs/en/query_language/create.md

```diff
@@ -299,4 +299,145 @@ Depending on dictionary [layout](dicts/external_dicts_dict_layout.md) one or mor
+- `IDENTIFIED WITH double_sha1_password BY 'qwerty'`
+- `IDENTIFIED WITH double_sha1_hash BY 'hash'`
+
+#### User Host
+
+User host is a host from which a connection to ClickHouse server could be established. Host can be specified in the `HOST` section of query by the following ways:
+
+- `HOST IP 'IP_ADDRESS_OR_SUBNETWORK'` — User can connect to ClickHouse server only from the specified IP address or a [subnetwork](https://en.wikipedia.org/wiki/Subnetwork). For example, `HOST IP '192.168.0.0/16'`.
+- `HOST ANY` — No limits.
+- `HOST LOCAL` — User can connect only locally.
+- `HOST NAME 'FQDN'` —  The user host can be specified as FQDN. For example, `HOST NAME 'mysite.com'`.
+- `HOST NAME REGEXP 'regexp'` — You can use [pcre](http://www.pcre.org/) regular expressions when specifying user hosts. For example, `HOST NAME REGEXP '.*\.mysite\.com'`.
+- `HOST LIKE 'te.mp.la.te'` — Allows you use the [LIKE](functions/string_search_functions.md#function-like) operator to filter the user hosts. For example, `HOST LIKE '%'` is equivalent to `HOST ANY`, or `HOST LIKE '192.168.%.%'` is equivalent to `HOST IP '192.168.0.0/16'`.
+
+Another way of specifying host is to use `@` syntax with the user name. Examples:
+
+- `CREATE USER mira@'127.0.0.1'` — Equivalent to the `HOST IP` syntax.
+- `CREATE USER mira@'localhost'` — Equivalent to the `HOST LOCAL` syntax.
+- `CREATE USER mira@'192.168.%.%'` — Equivalent to the `HOST LIKE` syntax.
+
+!!! info "Warning"
+    ClickHouse treats `user_name@'address'` as a user name as a whole. Thus, technically you can create multiple users with `user_name` and different constructions after `@`. We don't recommend to do so.
+
+
+### Example
+
+```sql
+CREATE USER mira HOST IP '127.0.0.1' IDENTIFIED WITH sha256_password BY 'qwerty'
+```
+
+This query creates the user account `mira` protected by the password `qwerty`. Also user should start client app at the host where the ClickHouse server runs.
+
+
+## CREATE ROLE {#create-role-statement}
+
+Creates a role.
+
+### Syntax
+
+```sql
+CREATE ROLE [IF NOT EXISTS | OR REPLACE] name
+    [SETTINGS variable [= value] [MIN [=] min_value] [MAX [=] max_value] [READONLY|WRITABLE] | PROFILE 'profile_name'] [,...]
+```
+
+### Description
+
+Role is a set of [privileges](grant.md#grant-privileges). A user assigned with a role gets all the privileges of this role. 
+
+A user can be assigned with multiple roles. Users can apply their assigned roles in arbitrary combinations by the [SET ROLE](misc.md#set-role-statement) statement. The final scope of privileges is a combined set of all the privileges of all the applied roles. If a user has privileges assigned directly to it's user account, they are also combined with the privileges granted by roles.
+
+User can have a default role which applies at user login. To set the default role, use the [SET DEFAULT ROLE](misc.md#set-default-role-statement) statement or the [ALTER USER](alter.md#alter-user-statement) statement.
```

Comment:
There can be multiple default roles, syntax:

```
SET DEFAULT ROLE role1, role2, ... TO user
```

You can also choose all granted roles to set as default:

```
SET DEFAULT ROLE ALL TO user
```

or none:
```
SET DEFAULT ROLE ALL TO user
```

or even more complex
```
SET DEFAULT ROLE ALL EXCEPT role1, role2 TO user
```

Reply:
`CREATE USER user DEFAULT ROLE role1, role2, ...` - grants roles and makes them default for the user
`ALTER USER user DEFAULT ROLE role1, role2, ...` - checks if the roles are granted; and if they're granted makes them default for the user (if they're not granted an exception is thrown)
`SET DEFAULT ROLE role1, role2, ... TO user` - checks if the roles are granted; and if they're granted makes them default for the user (if they're not granted an exception is thrown)

`ALTER USER DEFAULT ROLE` is almost synonym for `SET DEFAULT ROLE`.
If some of the specified roles are not granted to user `ALTER USER DEFAULT ROLE` will throw an exception as well as `SET DEFAULT ROLE`. `CREATE USER DEFAULT ROLE` is different, in that case it will grant specified roles and then make them default.